pex_reactor.go

```diff
@@ -61,31 +88,41 @@ func (pexR *PEXReactor) GetChannels() []*ChannelDescriptor {
 	}
 }
 
-// Implements Reactor
-func (pexR *PEXReactor) AddPeer(peer *Peer) {
-	// Add the peer to the address book
-	netAddr := NewNetAddressString(peer.ListenAddr)
-	if peer.IsOutbound() {
-		if pexR.book.NeedMoreAddrs() {
-			pexR.RequestPEX(peer)
+// AddPeer implements Reactor by adding peer to the address book (if inbound)
+// or by requesting more addresses (if outbound).
+func (r *PEXReactor) AddPeer(p *Peer) {
+	if p.IsOutbound() { // For outbound peers, the address is already in the books
+		if r.book.NeedMoreAddrs() {
+			r.RequestPEX(p)
 		}
-	} else {
-		// For inbound connections, the peer is its own source
-		// (For outbound peers, the address is already in the books)
-		pexR.book.AddAddress(netAddr, netAddr)
+	} else { // For inbound connections, the peer is its own source
+		addr := NewNetAddressString(p.ListenAddr)
+		r.book.AddAddress(addr, addr)
 	}
 }
 
-// Implements Reactor
-func (pexR *PEXReactor) RemovePeer(peer *Peer, reason interface{}) {
-	// TODO
+// RemovePeer implements Reactor by removing peer from the address book.
+//
+// The peer will be proposed to us by other peers (PexAddrsMessage) or himself
+// and we will add him again upon successful connection. Note that other peers
+// will remove him too. The peer will need to send first requests to others by
+// himself (he will have an addrbook or the seeds).
+func (r *PEXReactor) RemovePeer(p *Peer, reason interface{}) {
+	addr := NewNetAddressString(p.ListenAddr)
```

Comment:
AddPeer/RemovePeer are just housekeeping methods, we shouldn't remove a peer from the addrbook just because they got disconnected.

We could rename AddPeer/RemovePeer to "OnPeerConnect" and "OnPeerDisconnect".  If we aren't keeping track of local temp data for each peer here, then we don't have to do anything.